pkgs/development/python-modules/numpy/default.nix

```diff
@@ -54,6 +56,23 @@ in buildPythonPackage rec {
     # Patching of numpy.distutils is needed to prevent it from undoing the
     # patch to distutils.
     ./numpy-distutils-C++.patch
+    # f2py.f90mod_rules generates code with invalid function pointer conversions, which are
+    # clang 16 makes an error by default.
+    (fetchpatch {
+      url = "https://github.com/numpy/numpy/commit/609fee4324f3521d81a3454f5fcc33abb0d3761e.patch";
+      hash = "sha256-6Dbmf/RWvQJPTIjvchVaywHGcKCsgap/0wAp5WswuCo=";
+    })
+  ] ++ lib.optionals (lib.versionOlder version "1.25") [
+    # Backport from 1.25. `platform.machine` returns `arm64` on aarch64-darwin, which causes
+    # differing results between `_selected_real_kind_func` and Fortran’s `selected_real_kind`.
+    (fetchpatch {
+      url = "https://github.com/numpy/numpy/commit/afcedf4b63f4a94187e6995c2adea0da3bb18e83.patch";
+      hash = "sha256-cxBoimX5a9wC2qUIGAo5o/M2E9+eV63bV2/wLmfDYKg=";
+    })
+  ] ++ lib.optionals (stdenv.isDarwin && stdenv.isx86_64) [
```

Comment:
Please make this unconditional, none of the core python maintainers uses a Mac regularly to test numpy updates

Reply:
I can move the conditional to the Python-side, but it seems like the test should be run on platforms where it is expected to work. The patch itself just uses `pytest.mark.skipif` to skip the complex256 test on x86_64-darwin.

Reply:
Well, I see the complication that we already have lots of binaries on other platforms and this would make them rebuild again, slowing everything down.

Reply:
Then we can send a patch to staging afterwards making this unconditional

Reply:
Won’t the other patches cause rebuilds regardless? Am I making all of the changes conditional?